CMakeLists.txt

```diff
@@ -1,6 +1,6 @@
 cmake_minimum_required(VERSION 2.6)
 
-project(Effective_Modern_C++ CXX)
+project(Effective_Modern_C++ C CXX)
```

Comment:
Oki ... seems strange ... maybe a bug (fix) from CMake.
In my system (from official package 'cmake') in FindThreads.cmake i don't have in the script a test for switching between C and CXX project (like in: https://github.com/Kitware/CMake/blob/master/Modules/FindThreads.cmake around line 94-99).
Maybe the depot (for Ubuntu14.04/Mint) isn't synchronized with the depot of the last release of CMake.
Whatever, with my version (CMake 3.2.2) i need to to add C flag to permit the build of the test file 'CheckForPthreads.c' needed by CMake & 'FindThreads.cmake' to check if a PThread lib is available (with a default/system path).
With the last release of CMake (from github depot), it will not be necessary anymore.